frontend/server/controllers/ProblemController.php

```diff
@@ -899,37 +900,38 @@ private static function validateDetails(Request $r) {
         }
 
         try {
-            $r['problem'] = ProblemsDAO::getByAlias($r['problem_alias']);
+            $problem = ProblemsDAO::getByAlias($r['problem_alias']);
         } catch (Exception $e) {
             throw new InvalidDatabaseOperationException($e);
         }
 
-        if (is_null($r['problem'])) {
-            throw new NotFoundException('problemNotFound');
+        if (is_null($problem)) {
+            return null;
         }
 
         if (isset($r['statement_type']) && $r['statement_type'] != 'markdown') {
             throw new NotFoundException('invalidStatementType');
         }
 
         // If we request a problem inside a contest
-        if (self::validateProblemset($r)) {
-            if (!Authorization::isAdmin($r['current_identity_id'], $r['problemset'])) {
+        $problemset = self::validateProblemset($problem, $r['problemset_id'], $r['contest_alias']);
+        if (!is_null($problemset['problemset'])) {
```

Comment:
aquí debería ser:

```suggestion
        if (!is_null($problemset) && isset($problemset['problemset'])) {
```

porque `$problemset` puede ser `null`. y como `$problemset` es un arreglo (no un `Request`), no podemos usar `is_null` sino tenemos que usar `isset()`/`!empty()`.

Reply:
Listo